practices/java-jdbc/main/java/com/uabc/dowa/jdbc/repository/StudentRepository.java

```diff
@@ -1,23 +1,88 @@
 import com.uabc.dowa.jdbc.model.Student;
 
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
 import java.util.List;
 
 public class StudentRepository {
-    public void persistStudent(Student student) {
-
+    public void persistStudent(Student student) throws SQLException {
+        Connection conn = ConnectionFactory.getConnection();
+        PreparedStatement pstmt = null;
+        try {
+            pstmt = conn.prepareStatement("INSERT INTO student (first_name, last_name) VALUES ('"+student.getFirstName()+
+                    "','"+student.getLastName()+"')");
+            pstmt.execute();
+        } catch (SQLException e) {
+            System.err.println("SQLException: " + e.getMessage());
+        }
+        finally {
+            conn.close();
+        }
     }
 
-    public List<Student> findAllStudents() {
-        return null;
+    public List<Student> findAllStudents() throws SQLException {
+        Connection conn = ConnectionFactory.getConnection();
+        PreparedStatement pstmt = null;
+        ResultSet rs = null;
+        Student student = new Student();
+        List <Student> allStuds= new ArrayList<Student>();
+        pstmt = conn.prepareStatement("SELECT * FROM student");
+        try {
+            rs = pstmt.executeQuery();
+
+            while(rs.next()) {
+                student.setFirstName(rs.getString(2));
+                student.setLastName(rs.getString(3));
+                allStuds.add(student);
+            }
+        } catch (SQLException e) {
+            System.err.println("SQLException: " + e.getMessage());
+        }
+        finally {
+            conn.close();
+        }
+        return allStuds;
     }
 
-    public Student findStudentByFirstName(String firstName) {
-        return null;
+    public Student findStudentByFirstName(String firstName) throws SQLException{
+        Connection conn = ConnectionFactory.getConnection();
+        PreparedStatement pstmt = null;
+        ResultSet rs = null;
+        Student found = new Student();
+        try {
+            pstmt = conn.prepareStatement("SELECT * FROM student WHERE first_name = '" + firstName + "'");
+            rs = pstmt.executeQuery();
+            if(rs.next()) {
+                found.setFirstName(rs.getString(2));
+                found.setLastName(rs.getString(3));
+            }
+        } catch (SQLException e){
+            System.err.println("SQLException: " + e.getMessage());
+        }
+        finally {
+            conn.close();
+        }
+        return found;
     }
 
-    public void deleteStudent(Student student) {
+    public void deleteStudent(Student student) throws SQLException{
+        Connection conn = ConnectionFactory.getConnection();
+        PreparedStatement pstmt = null;
 
+        try {
+            pstmt = conn.prepareStatement("DELETE FROM student WHERE first_name = '"+student.getFirstName()+
```

Comment:
this is prone to sql injection, better use pstmt = conn.prepareStatement("DELETE FROM student WHERE first_name =  ?");
pstmt.setString(0, student.getFirstName()); //this code replaces the ? with the value internally and it also prevents sqlinjection